gstreamer-sdp/src/s_d_p_message.rs

```diff
@@ -8,4 +39,389 @@ impl SDPMessage {
+		unsafe {
+			from_glib(ffi::gst_sdp_message_attributes_to_caps(self.to_glib_none().0, caps.to_glib_none().0))
+		}
+	}
+
+	pub fn bandwidths_len(&self) -> u32 {
+		unsafe {
+			ffi::gst_sdp_message_bandwidths_len(self.to_glib_none().0)
+		}
+	}
+
+	pub fn dump(&self) -> SDPResult {
+		unsafe {
+			from_glib(ffi::gst_sdp_message_dump(self.to_glib_none().0))
+		}
+	}
+
+	pub fn emails_len(&self) -> u32 {
+		unsafe {
+			ffi::gst_sdp_message_emails_len(self.to_glib_none().0)
+		}
+	}
+
+	pub fn get_attribute(&self, idx: u32) -> Option<SDPAttribute> {
+		unsafe {
+			from_glib_none(ffi::gst_sdp_message_get_attribute(self.to_glib_none().0, idx))
+		}
+	}
+
+	pub fn get_attribute_val(&self, key: &str) -> Option<String> {
+		unsafe {
+			from_glib_none(ffi::gst_sdp_message_get_attribute_val(self.to_glib_none().0, key.to_glib_none().0))
+		}
+	}
+
+	pub fn get_attribute_val_n(&self, key: &str, nth: u32) -> Option<String> {
+		unsafe {
+			from_glib_none(ffi::gst_sdp_message_get_attribute_val_n(self.to_glib_none().0, key.to_glib_none().0, nth))
+		}
+	}
+
+	pub fn get_bandwidth(&self, idx: u32) -> Option<SDPBandwidth> {
+		unsafe {
+			from_glib_none(ffi::gst_sdp_message_get_bandwidth())
+		}
+	}
+
+	pub fn get_connection(&self) -> Option<SDPConnection> {
+		unsafe {
+			from_glib_none(ffi::gst_sdp_message_get_connection())
+		}
+	}
+
+	pub fn get_email(&self, idx: u32) -> Option<String> {
+		unsafe {
+			from_glib_none(ffi::gst_sdp_message_get_email(self.to_glib_none().0, idx))
+		}
+	}
+
+	pub fn get_information(&self) -> Option<String> {
+		unsafe {
+			from_glib_none(ffi::gst_sdp_message_get_information(self.to_glib_none().0))
+		}
+	}
+
+	pub fn get_key(&self) -> Option<SDPKey> {
+		unsafe {
+            from_glib_none(ffi::gst_sdp_message_get_key())
+        }
+	}
+
+	pub fn get_media(&self, idx: u32) -> Option<SDPMedia> {
+		unsafe {
+			from_glib_none(ffi::gst_sdp_message_get_media(self.to_glib_none().0, idx))
+		}
+	}
+
+	pub fn get_origin(&self) -> Option<SDPOrigin> {
+		unsafe {
+			from_glib_none(ffi::gst_sdp_message_get_origin())
+		}
+	}
+
+	pub fn get_phone(&self, idx: u32) -> Option<String> {
+		unsafe {
+			from_glib_none(ffi::gst_sdp_message_get_phone(self.to_glib_none().0, idx))
+		}
+	}
+
+	pub fn get_session_name(&self) -> Option<String> {
+		unsafe {
+			from_glib_none(ffi::gst_sdp_message_get_session_name(self.to_glib_none().0))
+		}
+	}
+
+	pub fn get_time(&self, idx: u32) -> Option<SDPTime> {
+		unsafe {
+			from_glib_none(ffi::gst_sdp_message_get_time())
+		}
+	}
+
+	pub fn get_uri(&self) -> Option<String> {
+		unsafe {
+			from_glib_none(ffi::gst_sdp_message_get_uri(self.to_glib_none().0))
+		}
+	}
+
+	pub fn get_version(&self) -> Option<String> {
+		unsafe {
+			from_glib_none(ffi::gst_sdp_message_get_version(self.to_glib_none().0))
+		}
+	}
+
+	pub fn get_zone(&self, idx: u32) -> Option<SDPZone> {
+		unsafe {
+			from_glib_none(ffi::gst_sdp_message_get_zone())
+		}
+	}
+
+	pub fn init(&mut self) -> SDPResult {
```

Comment:
This one here is actually the constructor

Reply:
That is, you call `glib_ffi::g_malloc(mem::size_of::<GstSDPMessage>())`, then `init` and then return that